File: android/appmanager/src/main/java/com/lmig/uscm/digital/appplatform/appmanager/AppManager.kt
package com.lmig.uscm.digital.appplatform.appmanager

import android.support.v4.content.LocalBroadcastManager

/**
 * App Manager
 *
 * Class representing the Application Manager
 *
 * @param [featureManager] An instance of a DSS [FeatureManager] component
 * @param [config] An instance of a DSS [AppConfig] component
 * @param [broadcastManager] An instance of a [LocalBroadcastManager]
 *
 * @property [navigationManager] An instance of a DSS [NavigationManager] component
 */
class AppManager(val featureManager: FeatureManager, val config: AppConfig, val broadcastManager: LocalBroadcastManager) {

    val navigationManager: NavigationManager

    /**
     * Initializes the App Manager
     */
    init {
        featureManager.mergeConfig(featureConfig = config.features)
        featureManager.setupViews(viewConfig = config.views)
        navigationManager = NavigationManager(featureManager = featureManager, broadcastManager = broadcastManager)
    }
}

File: ios/AppManager/AppManager.swift
//
//  AppManager.swift
//  AppManager
//
//  Copyright © 2017 Liberty Mutual. All rights reserved.
//
import UIKit

/// Main wrapper for all the features of the new platform
open class AppManager {
    public var featureManager: FeatureManager
    public var config: AppConfig
    public var navigationManager: NavigationManager

    /**
     Initialization method for the `AppManager`. After everything is set to local instance,
     it initiates default feature
     
     - parameter featureManager: Sets `FeatureManager` to the instance
     - parameter config: Sets `AppConfig` to the instance
     */
    public init (featureManager: FeatureManager, config: AppConfig) {
        self.featureManager = featureManager
        self.config = config

        self.featureManager.mergeConfig(featureConfig: self.config.features)
        self.featureManager.setupViews(viewConfig: self.config.views)
        self.navigationManager = NavigationManager(featureManager: self.featureManager)
    }
}
